File: src/src/main/hackerrank/helper/java/Scanner.java
package hackerrank.helper.java;

public class Scanner {

    private final InputStream _in;

    public Scanner(InputStream in) {
        _in = in;
    }

    public Integer nextInt() {
        return _in.getNextInt();
    }

    public String nextString() {
        return _in.getNextString();
    }

    public String nextLine() {
        return _in.getNextString();
    }

    public String next() {
        return this.nextString();
    }

    public void close() {
    }
}

File: src/src/main/scala/hackerrank/helper/Scanner.scala
package hackerrank.helper

class Scanner(in: InputStream) {
  def nextInt() = in.getNextInt()
  def nextString() = in.getNextString()
  def nextLine() = in.getNextString()
  def next() = in.getNextString()
  def close() = in.close()
}
